src/Avalonia.Controls/Button.cs

```diff
@@ -251,7 +253,10 @@ protected override void OnPointerReleased(PointerReleasedEventArgs e)
                 IsPressed = false;
                 e.Handled = true;
 
-                if (ClickMode == ClickMode.Release && new Rect(Bounds.Size).Contains(e.GetPosition(this)))
+                var hittest = this.GetVisualsAt(e.GetPosition(this));
+
+                if (ClickMode == ClickMode.Release &&
+                    hittest.Any(c => c == this || (c as IStyledElement)?.TemplatedParent == this) == true)
```

Comment:
Tiny tiny nit: the last `== true` isn't necessary.

Reply:
```suggestion
                    hittest.Any(c => c == this || (c as IStyledElement)?.TemplatedParent == this))
```

Reply:
haha yes this is some remaining of temp code lol

Reply:
should be fine now thanks